From bf482ae25d6971f705bc0369c9a71bd121380ba0 Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Mon, 5 May 2025 19:58:10 +0200
Subject: [PATCH 107/553] Introduce
 _qt_internal_android_get_platform_tools_path

Function returns the path to the Android platform tools. If the path is
changed in the future, this function will localize the required changes.

Pick-to: 6.5 6.8 6.9 6.10.0
Change-Id: Iea6e76c28e4260aaad6ba5429e2e46b0836c5e1a
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 6d027f547776c532d057fb51bc9ce34c5f9f28ab)
Reviewed-by: Assam Boudjelthia <assam.boudjelthia@qt.io>
---
 cmake/QtPlatformAndroid.cmake      | 3 ++-
 src/corelib/Qt6AndroidMacros.cmake | 8 +++++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/cmake/QtPlatformAndroid.cmake b/cmake/QtPlatformAndroid.cmake
index b480b1c14a6..60ed6f9e1c2 100644
--- a/cmake/QtPlatformAndroid.cmake
+++ b/cmake/QtPlatformAndroid.cmake
@@ -117,9 +117,10 @@ function(qt_internal_android_test_runner_arguments target out_test_runner out_te
     set(deployment_tool "${host_bin_dir}/androiddeployqt")
 
     _qt_internal_android_get_target_android_build_dir(android_build_dir ${target})
+    _qt_internal_android_get_platform_tools_path(platform_tools)
     set(${out_test_arguments}
         "--path" "${android_build_dir}"
-        "--adb" "${ANDROID_SDK_ROOT}/platform-tools/adb"
+        "--adb" "${platform_tools}/adb"
         "--skip-install-root"
         "--make" "\"${CMAKE_COMMAND}\" --build ${CMAKE_BINARY_DIR} --target ${target}_make_apk"
         "--apk" "${android_build_dir}/${target}.apk"
diff --git a/src/corelib/Qt6AndroidMacros.cmake b/src/corelib/Qt6AndroidMacros.cmake
index da124fe2ca9..503b217c425 100644
--- a/src/corelib/Qt6AndroidMacros.cmake
+++ b/src/corelib/Qt6AndroidMacros.cmake
@@ -1780,8 +1780,9 @@ function(_qt_internal_android_app_runner_arguments target out_runner_path out_ar
     set(${out_runner_path} "${runner_dir}/qt-android-runner.py" PARENT_SCOPE)
 
     _qt_internal_android_get_target_android_build_dir(android_build_dir ${target})
+    _qt_internal_android_get_platform_tools_path(platform_tools)
     set(${out_arguments}
-        "--adb" "${ANDROID_SDK_ROOT}/platform-tools/adb"
+        "--adb" "${platform_tools}/adb"
         "--build-path" "${android_build_dir}"
         "--apk" "${android_build_dir}/${target}.apk"
         PARENT_SCOPE
@@ -1941,5 +1942,10 @@ function(_qt_internal_android_find_asan_wrap_sh out_wrap_sh_path)
     endif()
 endfunction()
 
+# Returns the path to the Android platform-tools(adb is located there).
+function(_qt_internal_android_get_platform_tools_path out_var)
+    set(${out_var} "${ANDROID_SDK_ROOT}/platform-tools" PARENT_SCOPE)
+endfunction()
+
 set(QT_INTERNAL_ANDROID_TARGET_BUILD_DIR_SUPPORT ON CACHE INTERNAL
     "Indicates that Qt supports per-target Android build directories")
-- 
2.51.2

